src/gurt/telemetry.c

```diff
@@ -2361,7 +2366,12 @@ d_tm_allocate_shared_memory(int srv_idx, size_t mem_size)
 		return NULL;
 	}
 
-	return (uint64_t *)shmat(d_tm_shmid, NULL, 0);
+	addr = shmat(d_tm_shmid, NULL, 0);
+	if (addr == (void *)-1) {
+		D_ERROR("shmat failed, %s\n", strerror(errno));
```

Comment:
nit: this error message slightly inconsistent with the one above, here you just mention the function name that failed were above you elaborate, "unable to detach…" rather than "shmat…"

Reply:
I've made some updates there to elaborate and make them consistent.  See what you think.